sessions/nowcasting.qmd

```diff
@@ -31,6 +32,555 @@ The code in this session can be run as an interactive notebook using RStudio, or
+- The `reported_onsets` are the number of onsets that are reported on each day. This is calculated by multiplying the number of onsets by the proportion of onsets that are reported on each day. It has Poisson noise added to it to simulate the stochasticity in the reporting process.
+:::
+
+We can now fit our first nowcasting model. Here we assume exactly the same generative process as we used for simulation and model the number of onsets as independent draws from a normal distribution.
+
+```{r stan-simple-nowcast}
+mod <- cmdstan_model(here("stan", "simple-nowcast.stan"), include_paths = here("stan"))
+mod$print(line_numbers = TRUE)
+```
+
+::: {.callout-tip}
+## Take 2 minutes
+Familiarise yourself with the model above. What does it do?
+:::
+
+::: {.callout-note collapse="true"}
+## Solution
+- On line 2 we define a new function `condition_onsets_by_report.stan` which takes the number of onsets and reports and the delay distribution as input and returns the nowcasted number of onsets.
+- On line 17, this function is used to calculate the nowcasted number of onsets and this is then used in the likelihood.
+- On line 21, we define the generative process for the number of onsets. Here we assume that onsets are independent with each drawn from a normal distribution.
+:::
+
+Once again we can generate estimates from this model:
+
+```{r nowcast_fit}
+data <- list(
+  n = nrow(reported_onset_df) - 1,
+  obs = reported_onset_df$reported_onsets[-1],
+  report_max = length(proportion_reported) - 1,
+  report_cdf = proportion_reported 
+)
+simple_nowcast_fit <- mod$sample(
+  data = data, parallel_chains = 4, refresh = ifelse(interactive(), 50, 0), show_exceptions = FALSE, show_messages = FALSE
+)
+simple_nowcast_fit
+```
+
+We can now plot onsets alongside those nowcasted by the model:
+
+```{r simple-nowcast-onsets}
+nowcast_onsets <- simple_nowcast_fit |>
+  gather_draws(onsets[day]) |>
+  group_by(day, .variable) |>
+  summarise(
+    median = median(.value),
+    lower = quantile(.value, 0.05),
+    upper = quantile(.value, 0.95),
+    .groups = "drop"
+  ) |>
+  mutate(day = day + 1)
+```
+
+```{r plot_nowcast}
+reported_onset_df |> 
+  filter(day > 1) |>
+  left_join(nowcast_onsets, by = "day") |>
+  ggplot(aes(x = day, y = onsets)) +
+  geom_col() +
+  geom_ribbon(aes(ymin = lower, ymax = upper), alpha = 0.5) +
+  geom_line(aes(y = median))
+```
+
+::: {.callout-tip}
+As we found in the [using delay distributions to model the data generating process of an epidemic session](using-delay-distributions-to-model-the-data-generating-process-of-an-epidemic#estimating-a-time-series-of-infections), this simple model struggles to recreate the true number of onsets. This is because it does not capture the generative process of the data (i.e. the transmission process and delays from infection to onset). In the next section we will see how we can use a model that does capture this generative process to improve our nowcasts.
```

Comment:
it seems to work fairly well? This may need more motivation I think.

Reply:
Potentially it could also be a "going further" bit, i.e. if you've got it all try putting in the geometric random walk from before which should work better and see if it did.

Reply:
This was my point about all the text being true and then the data "changed" on a rerender and we had a decreasing outbreak -> model that works quite well. 

When you have a growing outbreak it should revert to the prior towards the end and hence look bad. 

The other related problem is that we know the delay distribution so the performance should be the best it can be

Reply:
might be worth adding a data point (as earlier)
<img width="712" alt="image" src="https://github.com/nfidd/nfidd/assets/1156307/2b0db979-d80d-49c8-800d-766f9cae931b">

Reply:
> might be worth adding a data point (as earlier)

?